include/realizations/catchment/Formulation_Constructors.hpp

```diff
@@ -18,7 +18,7 @@
 #include "Bmi_Multi_Formulation.hpp"
 #include "Bmi_Py_Formulation.hpp"
 #include <GenericDataProvider.hpp>
-#include "CsvPerFeatureForcingProvider.hpp"
+//#include "CsvPerFeatureForcingProvider.hpp"
```

Comment:
This change is actually what is causing your problem in `Catchment_Formulation.hpp`. This is the one place that `CsvPerFeatureForcingProvider.hpp` is needed, because the instances of the providers are instantiated in this class.

Reply:
You will need `#include "CsvPerFeatureForcingProvider.hpp"` in files where `CsvPerFeatureForcingProvider` is explicitly constructed... for the moment that means the two tshirt files also (though ideally those dependencies can be removed, per the discussion there).